### PR TITLE
Adds part 3 of the pipeline

### PR DESCRIPTION
**Pull request by Vlad-Dembrovskyi**
Adds processes:
- [x] `subset_sites_from_vcf`
- [x] `aggregate_all_sites`
- [x] `aggregate_all_mac`
- [x] `create_plink_set `

The wdl task `get_aggregate_and_site_file_list` was skipped because files were directly streamed from input to channel without a need to create an intermediate file, and because sites_file option was decided to be ommited for now. For same reason task `subset_sites_using_sites_file` was skipped.

### To test locally:
```
git clone https://github.com/lifebit-ai/avt.git
cd avt
git checkout vlad-adds-part-3
nextflow run main.nf \
          --coordinates_input_file 's3://lifebit-featured-datasets/projects/gel/avt/input_user_data/coordinates_input.tsv' \
          --phenotype_input_file 's3://lifebit-featured-datasets/projects/gel/avt/input_user_data/phenotypes_input.phe' \
          --aggregate_files_directory 's3://lifebit-featured-datasets/projects/gel/avt/vcfs' \
          --skip_vep_filtering true \
          --sparse_plink_name 's3://lifebit-featured-datasets/projects/gel/avt/plink/plink_dummy_file'
```
### Main CloudOS - tested successfully:
https://cloudos.lifebit.ai/public/jobs/5f8344118f81710113050d73
image
image

### GEL CloudOS - tested successfully:
image
image

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -251,6 +268,17 @@ ch_vep_severity_scale = !params.skip_vep_filtering ? Channel.fromPath(params.vep
...
+    .fromFilePairs("${params.aggregate_files_directory}/*{.vcf.gz,.vcf.gz.csi}", flat: true)
+    .into{ ch_vcf_to_count; ch_vcf_names }
+
+ch_mac_categories = Channel.from(params.mac_categories)
+
+ch_sparse_plink_files = Channel.from(params.sparse_plink_name)
+                                .map { name -> [file(name).getSimpleName(),
+                                                file(name+'.bed'),
+                                                file(name+'.bim'),
+                                                file(name+'.fam') ] }
```
Comment (cgpu) [inline main.nf:281]: beautiful usage of map, I like.

File: main.nf [hunk 2]
```diff
@@ -790,6 +823,185 @@ process create_regions_files {
...
+ * <--------------- Part 3 - GRM creation ---------------->
+ */
+
+
+// Creating a channel that will randomly sample a fraction of vcfs from initial vcf channel
+x = ch_vcf_to_count.count().val
+x = x * params.percent_chunks_to_keep.toInteger() / 100
+number_chunks_to_keep = Math.ceil(x).toInteger()
+
+ch_aggfiles_random_fraction = ch_vcf_names.randomSample(number_chunks_to_keep)
```
Comment (cgpu) [inline main.nf:836]: nice one 👍

File: main.nf [hunk 3]
```diff
@@ -790,6 +823,185 @@ process create_regions_files {
...
+    each file(platekeys) from ch_output_part_1_platekeys_2
+    each mac from ch_mac_categories
+
+    output:
+    tuple val(mac), file("${aggfile_name}_${mac[0]}_${mac[1]}_selected_sites.vcf.gz"), file("${aggfile_name}_${mac[0]}_${mac[1]}_selected_sites.vcf.gz.tbi") optional true into ch_files_to_aggregate
+
+    script:
+    mac1 = mac[0]
+    mac2 = mac[1]
+    variants_to_include = params.variants_to_include
```
Comment (cgpu) [inline main.nf:857]: `mac[0], mac[1]`

What are the contents of these vars?

**Review by cgpu — APPROVED** [review]
Lgtm 👍, feel free to squash this.

Only added some questions to make sure I am following the implementation logic, we can review in the final OR in part 4.

File: main.nf (hunk 3 above)
Comment (Vlad-Dembrovskyi) [reply main.nf:857]: mac1 and mac2 are minimum and maximum numbers of allele frequency to filter vcf. So each mac value is itself a tuple of two numbers - [ mac1, mac2]. 

Why I did this
The thing is that I need each vcf file to generate a filtered sub-vcf file for each possible filter. So for 3 vcfs and 7 mac filter pairs I need to have 21 outputs, 7 for each input vcf file. So I need to use each for mac values. Problem is, you can't use each with tuple, if I passed mac1 and mac 2 as separate elements of a tuple in channel. So I cheated - I pass one mac element, that is a tuple of two values [ [mac1,mac2] ] . So the channel is a series of one-values, just each value being a tuple :)  This allowed me to overcome missing each tuple functionality, or having to mix the two channels upstream.

**Pull request merged**